### PR TITLE
[SPARK-27476][SQL] Refactoring SchemaPruning rule to remove duplicate code

### PR DESCRIPTION
**Pull request by viirya**
## What changes were proposed in this pull request?

In SchemaPruning rule, there is duplicate code for data source v1 and v2. Their logic is the same and we can refactor the rule to remove duplicate code.

## How was this patch tested?

Existing tests.

### CODE REVIEW

**Comment by viirya** [comment]
cc @gengliangwang @cloud-fan

File: sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala [hunk 1]
```diff
@@ -1551,7 +1551,7 @@ object SQLConf {
       .internal()
       .doc("Prune nested fields from a logical relation's output which are unnecessary in " +
         "satisfying a query. This optimization allows columnar file format readers to avoid " +
-        "reading unnecessary nested column data. Currently Parquet and ORC v1 are the " +
```
Comment (gengliangwang) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala]: Nice catch

**Review by cloud-fan — APPROVED** [review]

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/SchemaPruning.scala [hunk 2]
```diff
@@ -50,73 +50,62 @@ object SchemaPruning extends Rule[LogicalPlan] {
...
-            op
-          }
-        } else {
-          op
-        }
+  private def prunePhysicalColumns(
+      output: Seq[AttributeReference],
+      projects: Seq[NamedExpression],
+      filters: Seq[Expression],
+      orgDataSchema: StructType,
```
Comment (gengliangwang) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/SchemaPruning.scala]: `orgDataSchema` is a bit confusing. How about just `dataSchema`.

File: sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/SchemaPruning.scala [hunk 3]
```diff
@@ -50,73 +50,62 @@ object SchemaPruning extends Rule[LogicalPlan] {
...
-
-            buildNewProjection(normalizedProjects, normalizedFilters, prunedRelationV2,
-              projectionOverSchema)
-          } else {
-            op
-          }
-        } else {
-          op
-        }
+  private def prunePhysicalColumns(
```
Comment (gengliangwang) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/SchemaPruning.scala]: Nit: Add comments about the return result.

**Review by gengliangwang — APPROVED** [review]
Overall LGTM. Thanks for the refactor.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Merged to master.
Thank you, @viirya , @cloud-fan , @gengliangwang .

**Pull request closed without merging**